components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -15,71 +15,304 @@ export class BitbucketServerApi {
             user,
-            `/${params.kind}/${params.userOrProject}/repos/${params.repositorySlug}/commits`,
+            `/${params.repoKind}/${params.owner}/repos/${params.repositorySlug}/commits${q}`,
         );
     }
 
-    getDefaultBranch(
+    async getDefaultBranch(
         user: User,
-        params: { kind: "projects" | "users"; userOrProject: string; repositorySlug: string },
+        params: { repoKind: "projects" | "users"; owner: string; repositorySlug: string },
     ): Promise<BitbucketServer.Branch> {
-        //https://bitbucket.gitpod-self-hosted.com/rest/api/1.0/users/jldec/repos/test-repo/default-branch
         return this.runQuery<BitbucketServer.Branch>(
             user,
-            `/${params.kind}/${params.userOrProject}/repos/${params.repositorySlug}/default-branch`,
+            `/${params.repoKind}/${params.owner}/repos/${params.repositorySlug}/default-branch`,
+        );
+    }
+
+    getHtmlUrlForBranch(params: {
+        repoKind: "projects" | "users";
+        owner: string;
+        repositorySlug: string;
+        branchName: string;
+    }): string {
+        return `https://${this.config.host}/${params.repoKind}/${params.owner}/repos/${
+            params.repositorySlug
+        }/browse?at=${encodeURIComponent(`refs/heads/${params.branchName}`)}`;
+    }
+
+    async getBranch(
+        user: User,
+        params: { repoKind: "projects" | "users"; owner: string; repositorySlug: string; branchName: string },
+    ): Promise<BitbucketServer.BranchWithMeta> {
+        const result = await this.runQuery<BitbucketServer.Paginated<BitbucketServer.BranchWithMeta>>(
+            user,
+            `/${params.repoKind}/${params.owner}/repos/${params.repositorySlug}/branches?details=true&filterText=${params.branchName}&boostMatches=true`,
+        );
+        const first = result.values && result.values[0];
+        if (first && first.displayId === params.branchName) {
+            first.latestCommitMetadata =
+                first.metadata["com.atlassian.bitbucket.server.bitbucket-branch:latest-commit-metadata"];
+            first.htmlUrl = this.getHtmlUrlForBranch({
+                repoKind: params.repoKind,
+                owner: params.owner,
+                repositorySlug: params.repositorySlug,
+                branchName: first.displayId,
+            });
+            return first;
+        }
+        throw new Error(`Could not find branch "${params.branchName}."`);
+    }
+
+    async getBranches(
+        user: User,
+        params: { repoKind: "projects" | "users"; owner: string; repositorySlug: string },
+    ): Promise<BitbucketServer.BranchWithMeta[]> {
+        const result = await this.runQuery<BitbucketServer.Paginated<BitbucketServer.BranchWithMeta>>(
+            user,
+            `/${params.repoKind}/${params.owner}/repos/${params.repositorySlug}/branches?details=true&orderBy=MODIFICATION&limit=1000`,
+        );
```

Comment:
We seem to be crafting a lot of these URLs ourselves where an SDK could simplify our lives a bit. I found https://www.npmjs.com/package/bitbucket but haven't tried it.

Reply:
"bitbucket.org" is not "bitbucket server", unfortunately. there is nothing out that which covers what we need.